homeassistant/components/onewire/sensor.py

```diff
@@ -249,9 +326,9 @@ def get_entities(onewirehub: OneWireHub, config):
 class OneWireDirect(OneWire):
     """Implementation of a 1-Wire sensor directly connected to RPI GPIO."""
 
-    def __init__(self, name, device_file, sensor_type, device_info, owsensor):
+    def __init__(self, name, device_file, device_info, owsensor):
         """Initialize the sensor."""
-        super().__init__(name, device_file, sensor_type, device_info)
+        super().__init__(name, device_file, "temperature", "Temperature", device_info)
         self._owsensor = owsensor
```

Comment:
So it never can be something else than a temperature entity in the future ?

Reply:
For now, OneWireDirect (SysBus) only supports temperature sensors.
Extra functionnality is currently only available through OneWireProxy (OWServer)